core/lib/types/src/l2_to_l1_log.rs

```diff
@@ -61,14 +66,46 @@ pub fn l2_to_l1_logs_tree_size(protocol_version: ProtocolVersionId) -> usize {
     pub const PRE_BOOJUM_L2_L1_LOGS_TREE_SIZE: usize = 512;
     pub const VM_1_4_0_L2_L1_LOGS_TREE_SIZE: usize = 2048;
     pub const VM_1_4_2_L2_L1_LOGS_TREE_SIZE: usize = 4096;
+    pub const VM_1_5_0_L2_L1_LOGS_TREE_SIZE: usize = 16384;
 
     if protocol_version.is_pre_boojum() {
         PRE_BOOJUM_L2_L1_LOGS_TREE_SIZE
     } else if protocol_version.is_1_4_0() || protocol_version.is_1_4_1() {
         VM_1_4_0_L2_L1_LOGS_TREE_SIZE
-    } else {
+    } else if protocol_version.is_1_4_2() {
         VM_1_4_2_L2_L1_LOGS_TREE_SIZE
+    } else {
+        VM_1_5_0_L2_L1_LOGS_TREE_SIZE
+    }
+}
+
+/// Returns the blob hashes parsed out from the system logs
+pub fn parse_system_logs_for_blob_hashes(
+    protocol_version: &ProtocolVersionId,
+    system_logs: &[SystemL2ToL1Log],
+) -> Vec<H256> {
+    let num_required_blobs = num_blobs_required(protocol_version) as u32;
+    let num_created_blobs = num_blobs_created(protocol_version) as u32;
+
+    if num_created_blobs == 0 {
+        return vec![H256::zero(); num_required_blobs as usize];
     }
```

Comment:
I think this `if` not required

Reply:
we need it because if we're in a protocol version that doesnt generate the linear hash logs, we can mistakenly use the expected system upgrade txn hash

Reply:
Hmm, I mean we won't use system upgrade log because this
```
log.0.key >= H256::from_low_u64_be(BLOB1_LINEAR_HASH_KEY as u64)
                && log.0.key
                    < H256::from_low_u64_be((BLOB1_LINEAR_HASH_KEY + num_created_blobs) as u64)
```
will never be true when `num_created_blobs == 0`. But I don't insist on removing the return, agree that in theory if we change method impl and it doesn't work for `num_created_blobs == 0` then the if will save us.